storage/trace.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"context"
+	"os"
+
+	"cloud.google.com/go/storage/internal"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	otelcodes "go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/trace"
+)
+
+const (
+	storageOtelTracingDevVar = "GO_STORAGE_DEV_OTEL_TRACING"
+	defaultTracerName        = "cloud.google.com/go/storage"
+	gcpClientRepo            = "googleapis/google-cloud-go"
+	gcpClientArtifact        = "storage"
+)
+
+// isOTelTracingDevEnabled checks the development flag until experimental feature is launched.
+func isOTelTracingDevEnabled() bool {
```

Comment:
This func seems unused? What is the actual purpose of the env var?

Reply:
It will be used during development to enable the new stuff only if that var is set. It will be removed once launched

Reply:
Yes, but it doesn't actually seem used anywhere currently, in which case it should be removed from this PR.

Reply:
`TestTraceStorageTraceStartEndSpan` uses this development flag env var.  

It verifies the storage trace package works only when this development flag is enabled (so we're not breaking existing usage). It also tests that the new startSpan accepts StartSpanOption and returns a span so attributes can be easily set through a library call.